pyproject.toml

```diff
@@ -25,14 +25,16 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
+python = ">=3.8,<3.13"
 numpy = [
     { version = ">=1.19", python = ">=3.8,<3.11" },
-    { version = ">=1.23", python = ">=3.11" },
+    { version = ">=1.23", python = ">=3.11,<3.12" },
+    { version = ">=1.26", python = ">=3.12" },
 ]
 pandas = [
     { version = ">=1.1,<1.6", python = ">=3.8,<3.11" },
-    { version = ">=1.5,<1.6", python = ">=3.11" },
+    { version = ">=1.5,<1.6", python = ">=3.11,<3.12" },
+    { version = ">=2.1.1", python = ">=3.12" },
```

Comment:
Did we validate pandas 2+ works? Particularly with datasets where DataFrames are used heavily

Reply:
Successfully ran dataset integration tests on Python 3.12 with Pandas 2.2
Do we want to pin the python 3.12 pandas to `^2.1.1` so we don't pull in [pandas 3](https://pandas.pydata.org/docs/whatsnew/v2.2.0.html#upcoming-changes-in-pandas-3-0)?

Reply:
Good call — going to pin to `<3`. Noting personal preference that I generally prefer that notation to `^` notation for clarity.